src/components/DashboardHome/MainContent.jsx

```diff
@@ -1,58 +1,93 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import MediaQuery from 'react-responsive';
-import { breakpoints } from '@edx/paragon';
+import { StaticQuery, graphql } from 'gatsby';
+import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { breakpoints, StatusAlert } from '@edx/paragon';
 
 import CourseSection from '../CourseSection/CourseSection';
 import InProgressCourseCard from '../CourseCard/InProgressCourseCard';
 import UpcomingCourseCard from '../CourseCard/UpcomingCourseCard';
 import CompletedCourseCard from '../CourseCard/CompletedCourseCard';
 import Sidebar from './Sidebar';
 
-import sampleApiResponse from './sampleApiResponse';
+import {
+  clearProgramEnrollmentOverview,
+  fetchProgramEnrollmentOverview,
+} from '../../data/actions/programEnrollments';
+
+const MainContentQuery = graphql`
+  query {
+    site {
+      siteMetadata {
+        programUUID
+      }
+    }
+  }
+`;
 
 class MainContent extends Component {
-  state = {
-    courses: {
+  componentDidMount() {
+    const { programUUID } = this.props;
+    this.props.fetchProgramEnrollmentOverview(programUUID);
+  }
+
+  componentWillUnmount() {
+    this.props.clearProgramEnrollmentOverview();
+  }
+
+  groupCourseEnrollmentsByStatus = () => {
+    const { courseRuns } = this.props;
+
+    const enrollments = {
       'in-progress': [],
```

Comment:
do we want to update this to `in_progress` for consistency? up to you

Reply:
Yeah. The API comes back with the status as `in_progress`. Will rename this key to match.